opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/RevealJsonFormFragmentPresenter.java

```diff
@@ -49,9 +50,11 @@ public RevealJsonFormFragmentPresenter(JsonFormFragment formFragment, JsonFormIn
     @Override
     public ValidationStatus writeValuesAndValidate(LinearLayout mainView) {
         ValidationStatus validationStatus = super.writeValuesAndValidate(mainView);
+        boolean hasMapView = false;
         if (validationStatus.isValid()) {
             for (View childAt : formFragment.getJsonApi().getFormDataViews()) {
                 if (childAt instanceof RevealMapView) {
+                    hasMapView = true;
                     RevealMapView mapView = (RevealMapView) childAt;
                     validationStatus = GeoWidgetFactory.validate(formFragment, mapView);
                     if (validationStatus.isValid()) {
```

Comment:
Seems like this needs to be tested whether or not a mapview exists.

Better put it outside the parent if statement?

Reply:
The default is `false` when the variable is defined. So its put inside the if statement so that its set to true if the form a mapview.

There is no need of testing for false since its the  default

Reply:
OK. I see the conditionals test different things.